src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,649 @@
+   *
+   * You can have overlapping endpoint IP addresses for file systems deployed in the same VPC/route tables, as long as they don't overlap with any subnet.
+   *
+   * @default - an unused IP address range from the 198.19.* range
+   */
+  readonly endpointIpAddressRange?: string;
+
+  /**
+   * The ONTAP administrative password for the `fsxadmin` user with which you administer your file system using the NetApp ONTAP CLI and REST API.
+   *
+   * If you don't specify a password, Amazon FSx will not set one. In that case, the user will not be able to log in.
+   *
+   * You can change the admin password at any time through the management console.
+   *
+   * @default - do not set an admin password
+   */
+  readonly fsxAdminPassword?: SecretValue;
+
+  /**
+   * How many high-availability (HA) pairs of file servers will power your file system.
+   *
+   * First-generation file systems are powered by 1 HA pair.
+   * Second-generation multi-AZ file systems are powered by 1 HA pair.
+   * Second generation single-AZ file systems are powered by up to 12 HA pairs.
+   *
+   * The value of this property affects the values of `storageCapacity`, `iops`, and `throughputCapacity`.
+   *
+   * Block storage protocol support (iSCSI and NVMe over TCP) is disabled on file systems with more than 6 HA pairs.
+   *
+   * @see https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/administering-file-systems.html#HA-pairs
+   * @see https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/supported-fsx-clients.html#using-block-storage
+   *
+   * @default 1
+   */
+  readonly haPairs?: number;
+
+  /**
+   * The subnet in which you want the preferred file server to be located.
+   *
+   * This value is required when `deploymentType` is set to `MULTI_AZ_1` or `MULTI_AZ_2`.
+   *
+   * @default - no default value (This value is not used for single-AZ file systems, but it is required for multi-AZ file systems)
+   */
+  readonly preferredSubnet?: aws_ec2.ISubnet;
+
+  /**
+   * The route tables in which Amazon FSx creates the rules for routing traffic to the correct file server.
+   *
+   * You should specify all virtual private cloud (VPC) route tables associated with the subnets in which your clients are located.
+   *
+   * Amazon FSx manages VPC route tables for Multi-AZ file systems using tag-based authentication.
+   * These route tables are tagged with Key: AmazonFSx; Value: ManagedByAmazonFSx.
+   *
+   * @see https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/unable-to-access.html#vpc-route-tables-not-tagged
+   *
+   * @default - Amazon FSx selects your VPC's default route table.
+   */
+  readonly routeTables?: aws_ec2.IRouteTable[];
+
+  /**
+   * The throughput capacity per HA pair for the file system.
+   *
+   * For SINGLE_AZ_1 and MULTI_AZ_1 file systems, valid values are 128, 256, 512, 1024, 2048, or 4096 MBps.
+   * For SINGLE_AZ_2, valid values are 1536, 3072, or 6144 MBps.
+   * For MULTI_AZ_2, valid values are 384, 768, 1536, 3072, or 6144 MBps.
+   *
```

Comment:
It is forgotten to maintain it in the future when the number of types increases and users can know about it in the enum-like class. In other words, this is good to remove it.